### PR TITLE
MAT-10054 generate composite measure bundle

### PR DESCRIPTION
**Pull request by adongare**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-10054](https://jira.cms.gov/browse/MAT-10054)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java [hunk 1]
```diff
@@ -66,41 +66,42 @@ public Bundle createMeasureBundle(
...
         new Bundle().setType(Bundle.BundleType.TRANSACTION).addEntry(measureEntryComponent);
     log.info("Measure bundle entry created successfully {}", madieMeasure.getId());
-    // Bundle entries for all the library resources of a MADiE Measure
-    List<Bundle.BundleEntryComponent> libraryEntryComponents =
-        createBundleComponentsForLibrariesOfMadieMeasure(
-            expressions, madieMeasure, bundleType, errorSeverity, accessToken);
-    libraryEntryComponents.forEach(bundle::addEntry);
-    log.info("Included library components created successfully {}", madieMeasure.getId());
+    // skip included libraries and effective data requirements for composite measure
+    if (!madieMeasure.getMeasureMetaData().isComposite()) {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:70]: Style preference: short circuit guard check vs large if.
```suggestion
    if (madieMeasure.getMeasureMetaData().isComposite()) {
        return bundle;
    }
```

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:70]: The `if` on line 99-105 can be moved above this `if` so that only the non-composite logic runs when the short-circuit is passed.

File: src/main/java/gov/cms/madie/madiefhirservice/constants/UriConstants.java [hunk 2]
```diff
@@ -97,6 +100,10 @@ public static final class CqfMeasures {
 
     public static final String IMPROVEMENT_NOTATION_GUIDANCE_URI =
         "http://hl7.org/fhir/StructureDefinition/cqf-improvementNotationGuidance";
+    public static final String CQFM_GROUP_ID_URI =
+        "http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-groupId";
+    public static final String CQM_COMPONENT_URI =
+        "http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-component";
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/constants/UriConstants.java]: Variable name and string are slightly mismatched. Double checking this is expected.
CQ**M**_COMPONENT vs cq**fm**-component.

File: pom.xml [hunk 3]
```diff
@@ -181,7 +181,7 @@
 		<dependency>
 			<groupId>gov.cms.madie</groupId>
 			<artifactId>madie-java-models</artifactId>
-			<version>0.9.1-SNAPSHOT</version>
+			<version>0.9.6-SNAPSHOT</version>
```
Comment (jkotanchik-SB) [inline pom.xml:184]: Why the jump from .1 to .6?

Comment (adongare) [reply pom.xml:184]: I think it hasn't been updated for a while.

File: src/main/java/gov/cms/madie/madiefhirservice/constants/UriConstants.java (hunk 2 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/constants/UriConstants.java]: updated it.

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:70]: could have done that but for non-composite measures we need to prepare the effective data requirements before HR generation.

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:70]: drat. oh well.

File: pom.xml (hunk 3 above)
Comment (jkotanchik-SB) [reply pom.xml:184]: Oh, never mind. I thought this was the service version, and not a dependency.

**Review by jkotanchik-SB — APPROVED** [review]
